src/platform.ts

```diff
@@ -0,0 +1,81 @@
+/* --------------------------------------------------------------------------------------------
+ * Copyright (c) Microsoft Corporation. All Rights Reserved.
+ * See 'LICENSE' in the project root for license information.
+ * ------------------------------------------------------------------------------------------ */
+
+import * as os from 'os';
+import * as util from './util';
+import { LinuxDistribution } from './linuxDistribution';
+
+function platformToJreOs(platform: string) {
+    return {
+        'win32': 'windows',
+        'linux': 'linux',
+        'darwin': 'mac'
+    }[platform];
+}
+
+function archToJreArch(arch: string) {
+    return {
+        'x86_64': 'x64',
+        'x86': 'x32'
+    }[arch];
+}
+
+export class PlatformInformation {
+    constructor(public os: string, public arch: string) { }
+
+    public static GetPlatformInformation(): Promise<PlatformInformation> {
+        const platform: string = os.platform();
+        let architecturePromise: Promise<string>;
+
+        switch (platform) {
+            case 'win32':
+                architecturePromise = PlatformInformation.GetWindowsArchitecture();
+                break;
+
+            case 'linux':
+            case 'darwin':
+                architecturePromise = PlatformInformation.GetUnixArchitecture();
+                break;
+        }
+
+        return architecturePromise.then(arch => {
+            return new PlatformInformation(platformToJreOs(platform), archToJreArch(arch));
+        });
+    }
+
+    public static GetUnknownArchitecture(): string { return 'Unknown'; }
+
+    private static async GetWindowsArchitecture(): Promise<string> {
```

Comment:
Nitpicking: You define this as an `async` function, but there isn't any `await`. Is that intended?